### PR TITLE
[[ Docs ]] Improve variadic syntax declarations

### PR DESCRIPTION
**Pull request by montegoulding**
The variadic syntax declarations were not very consisten with the
way they used `...` to describe addition optional parameters of the
same syntax. This patch attempts to make them consistent.

### CODE REVIEW

File: docs/dictionary/function/libUrlFormData.lcdoc [hunk 1]
```diff
@@ -2,7 +2,7 @@ Name: libURLFormData
 
 Type: function
 
-Syntax: libURLFormData(<key_1>, <value_1>, ..., <key_n>, <value_n>)
+Syntax: libURLFormData(<key1>, <value1> [, <key2>, <value2> ...])
```
Comment (livecodeali) [inline docs/dictionary/function/libUrlFormData.lcdoc:5]: shouldn't this be 
```
libURLFormData(<key1>, <value1> [, <key2>, <value2> [, ...]])
```
for consistency with the other ones?

File: docs/dictionary/function/revOpenDatabase.lcdoc [hunk 2]
```diff
@@ -3,15 +3,15 @@ Name: revOpenDatabase
...
+Syntax: revOpenDatabase(<databaseType>, {<hostAddress>, <databaseName>, [<userName>],[<passWord>],{[<useSSL>], [<socketAddr>], [<rwTimeout>], [<autoReconnect>] | [<cursorType>] | [<sslOption> [,...]]} | <filePath>,[<sqliteOptions>]})
 
 Syntax: revOpenDatabase("mysql", <hostAddress>, <databaseName>, [<userName>],[<passWord>],[<useSSL>], [<socketAddr>], [<rwTimeout>], [<autoReconnect>])
 
 Syntax: revOpenDatabase("odbc", <hostAddress>, <databaseName>, [<userName>],[<passWord>],[<cursorType>])
 
 Syntax: revOpenDatabase("sqlite",<filePath>,[<sqliteOptions>])
 
-Syntax: revOpenDatabase("postgresql", <hostAddress>, <databaseName>, [<userName>], [<passWord>], [<sslOption>],...)
+Syntax: revOpenDatabase("postgresql", <hostAddress>, <databaseName>, [<userName>], [<passWord>], [<sslOption> [,...]])
```
Comment (livecodeali) [inline docs/dictionary/function/revOpenDatabase.lcdoc]: We should probably be consistent with a space between `,` and `...` too

File: docs/dictionary/function/libUrlFormData.lcdoc (hunk 1 above)
Comment (montegoulding) [reply docs/dictionary/function/libUrlFormData.lcdoc:5]: Hmm... I'm not sure as `[, ...]` to me means repeat the last thing with a comma... where there's more complex repetitions such as here and `or type <type> ...` I think I it should wrap the whole thing

File: docs/dictionary/function/revOpenDatabase.lcdoc (hunk 2 above)
Comment (montegoulding) [reply docs/dictionary/function/revOpenDatabase.lcdoc]: ok, yes... it should be `[<sslOption>[ ,...]]`

File: docs/dictionary/function/libUrlFormData.lcdoc (hunk 1 above)
Comment (livecodeali) [reply docs/dictionary/function/libUrlFormData.lcdoc:5]: But then the `or` example to be consistent with this should be
```
ask file <prompt> [with <defaultFilePath>] [with type <types> [or type <types> ...]] [as sheet]
```

Comment (montegoulding) [reply docs/dictionary/function/libUrlFormData.lcdoc:5]: ah... yes it should... I should revert those two I think... woops. Or do you think I'm wrong here and right there ;-)

Comment (livecodeali) [reply docs/dictionary/function/libUrlFormData.lcdoc:5]: I don't know!

Comment (livecodeali) [reply docs/dictionary/function/libUrlFormData.lcdoc:5]: Actually, is there not some kind of notation for `(this block can appear 0 or more times)`

Comment (runrevmark) [reply docs/dictionary/function/libUrlFormData.lcdoc:5]: Was this ever consistent? For LCB syntax definitions we use
```
{ <thing> } -- repeat 0 or more times
{ <thing> , <separator> } -- repeat 0 or more times, separating with <separator> }
```
What do we want '...' to mean? :)

File: docs/dictionary/command/answer-file-with-type.lcdoc [hunk 3]
```diff
@@ -2,7 +2,7 @@ Name: answer file with type
 
 Type: command
 
-Syntax: answer file[s] <prompt> [with <defaultPath>] [with type <types> [or type <types> ...]] [titled <windowTitle>] [as sheet]
+Syntax: answer file[s] <prompt> [with <defaultPath>] [with type <types> [or type <types> [or ...]]] [titled <windowTitle>] [as sheet]
```
Comment (montegoulding) [inline docs/dictionary/command/answer-file-with-type.lcdoc]: revert as previous was correct

File: docs/dictionary/command/ask-file-with-type.lcdoc [hunk 4]
```diff
@@ -2,7 +2,7 @@ Name: ask file with type
 
 Type: command
 
-Syntax: ask file <prompt> [with <defaultFilePath>] [with type <types> [or type <types> ...]] [as sheet]
+Syntax: ask file <prompt> [with <defaultFilePath>] [with type <types> [or type <types> [or ...]]] [as sheet]
```
Comment (montegoulding) [inline docs/dictionary/command/ask-file-with-type.lcdoc]: as above

File: docs/dictionary/command/mobileControlDo.lcdoc [hunk 5]
```diff
@@ -4,7 +4,7 @@ Synonyms: iphonecontroldo
 
 Type: command
 
-Syntax: mobileControlDo <idOrName>, <action>, ...action specific parameters...
+Syntax: mobileControlDo <idOrName>, <action> [, <actionParameter> [, ...]]
```
Comment (montegoulding) [inline docs/dictionary/command/mobileControlDo.lcdoc]: should be:
```
Syntax: mobileControlDo <idOrName>, <action> [, <actionParameter> ...]
```

File: docs/dictionary/function/revBrowserCallScript.lcdoc [hunk 6]
```diff
@@ -2,7 +2,7 @@ Name: revBrowserCallScript
 
 Type: function
 
-Syntax: revBrowserCallScript(<instanceId>, <functionName> [, <parameter1>, <parameter2>, ..., <parameterN>])
+Syntax: revBrowserCallScript(<instanceId>, <functionName> [, <functionParameter> [, ...]])
```
Comment (montegoulding) [inline docs/dictionary/function/revBrowserCallScript.lcdoc]: Should be:
```
Syntax: revBrowserCallScript(<instanceId>, <functionName> [, <functionParameter> ...])
```

File: docs/dictionary/function/revOpenDatabase.lcdoc (hunk 2 above)
Comment (montegoulding) [reply docs/dictionary/function/revOpenDatabase.lcdoc]: These should be:
```
revOpenDatabase("postgresql", <hostAddress>, <databaseName>, [<userName>], [<passWord>] [, <sslOption> ...])
```

File: docs/dictionary/function/revOpenDatabase.lcdoc [hunk 7]
```diff
@@ -3,15 +3,15 @@ Name: revOpenDatabase
 Synonyms: revdb_connect
 
 Type: function
-Syntax: revOpenDatabase(<databaseType>, {<hostAddress>, <databaseName>, [<userName>],[<passWord>],{[<useSSL>], [<socketAddr>], [<rwTimeout>], [<autoReconnect>] | [<cursorType>] | [<sslOption>],...} | <filePath>,[<sqliteOptions>]})
+Syntax: revOpenDatabase(<databaseType>, {<hostAddress>, <databaseName>, [<userName>],[<passWord>], {[<useSSL>], [<socketAddr>], [<rwTimeout>], [<autoReconnect>] | [<cursorType>] | [<sslOption> ...]} | <filePath>,[<sqliteOptions>]})
```
Comment (montegoulding) [inline docs/dictionary/function/revOpenDatabase.lcdoc]: @livecodeali looking at this I'm thinking that the other syntax definitions below actually fully cover this one? Perhaps this one can be removed given how differently the function behaves depending on the database type I think it makes sense to list them separately.

Comment (livecodeali) [reply docs/dictionary/function/revOpenDatabase.lcdoc]: Agreed

**Comment by livecodeali** [comment]
@livecode-vulcan review ok fe5c8c3

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok fe5c8c321bdd238a3a6b56d79d76a575be928acb

**Comment by livecode-vulcan** [comment]
😞 test failure fe5c8c321bdd238a3a6b56d79d76a575be928acb

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/787)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/803)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/777)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/721)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/724)
* try-community-universal-mac-macosx10.6: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/781)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/803)
* try-community-x86-linux-debian7: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/786)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/803)
* try-community-x86_64-linux-debian7: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/785)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/796)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/817)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/781)

**Comment by livecodeali** [comment]
@livecode-vulcan retry fe5c8c3

**Comment by livecode-vulcan** [comment]
😞 test failure fe5c8c321bdd238a3a6b56d79d76a575be928acb

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/788)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/804)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/778)
* try-community-universal-ios-iphoneos10.3: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/722)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/725)
* try-community-universal-mac-macosx10.6: [pending](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/782)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/804)
* try-community-x86-linux-debian7: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/787)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/804)
* try-community-x86_64-linux-debian7: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/786)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/797)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/818)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/782)

**Comment by montegoulding** [comment]
@livecode-vulcan retry fe5c8c3

**Comment by livecode-vulcan** [comment]
😎 test success fe5c8c321bdd238a3a6b56d79d76a575be928acb

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/792)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/808)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/782)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/726)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/729)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/786)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/808)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/791)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/808)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/790)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/801)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/822)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/786)

**Pull request merged**